src/main/java/nextstep/security/authentication/OAuthLoginRedirectFilter.java

```diff
@@ -0,0 +1,65 @@
+package nextstep.security.authentication;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import nextstep.app.security.Oauth2LoginProperties;
+import nextstep.security.authentication.domain.OAuthLoginRedirectRequest;
+
+import java.io.IOException;
+
+public class OAuthLoginRedirectFilter implements Filter {
+
+    private static final String OAUTH2_LOGIN_REDIRECT_METHOD = "GET";
+    private static final String OAUTH2_LOGIN_REDIRECT_URI = "/oauth2/authorization/";
+
+    private final Oauth2LoginProperties oauth2LoginProperties;
+
+    public OAuthLoginRedirectFilter(Oauth2LoginProperties oauth2LoginProperties) {
+        this.oauth2LoginProperties = oauth2LoginProperties;
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+        final HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+
+        final String httpMethod = httpServletRequest.getMethod();
+        final String requestUri = httpServletRequest.getRequestURI();
+
+        if (!isOauth2Login(httpMethod, requestUri)) {
+            chain.doFilter(request, response);
+            return;
+        }
+
+        final String provider = requestUri.replace(OAUTH2_LOGIN_REDIRECT_URI, "");
+
+        Oauth2LoginProperties.OAuth2Provider oAuth2Provider = oauth2LoginProperties.getProvider(provider);
+
+        if (oAuth2Provider == null) {
+            chain.doFilter(request, response);
+        }
```

Comment:
chain.doFilter(request, response); 호출 이후에도 코드가 실행될 가능성이 있네요 🤔 

```suggestion
        if (oAuth2Provider == null) {
            chain.doFilter(request, response);
            return;
        }
```

Reply:
헉! 놓쳤던 실수였던 것 같습니다 ㅠ.ㅠ 018b1eaef4504edf549395f0db99d484817ee91e 커밋으로 return을 추가하였습니다